File: code/q1859.js
let s = "is2 sentence4 This1 a3";
console.log(s);
s = s.split(" ");
console.log(s);
let store = {};
let arr = [];
for (let i of s) {
    arr.push(i[i.length - 1]);
    store[i[i.length - 1]] = i.slice(0, i.length - 1)
}
console.log(store);
arr.sort((a, b) => {
    return a - b;
});
console.log(arr);
let res = "";
for (let i of arr) {
    console.log(i, store[i]);
    res += store[i] + " ";
}
res = res.slice(0, -1);
console.log(res);

File: code/q1859.py
#1859. Sorting the Sentence
s = "is2 sentence4 This1 a3"
print(s)
s = s.split(" ")
print(s)
store = {}
arr = []
for i in s:
    arr.append(i[-1])
    store[i[-1]] = i[0:len(i)-1]
arr.sort()
res = ""
for i in arr:
    res += store[i] + " "
res = res[:len(res)-1]
print(res)

